### PR TITLE
Release Azure.ResourceManager.NewRelicObservability for first GA

### LINKED ISSUE #36747: Release Azure.ResourceManager.NewRelicObservability first GA
### Library name

Azure.ResourceManager.NewRelicObservability

### Please describe the feature.

As title

### PR DESCRIPTION
**Pull request by vipray-ms**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/36747

Adding Stable Swagger Version for NewRelicObservability for **GA version SDK release**.

- Added tests
- updated readme and files for stable swagger version.

### CODE REVIEW

**Comment by vipray-ms** [comment]
Hey @live1206,
Please release GA .NET SDK version. Also please add the documentation here.
https://learn.microsoft.com/en-us/dotnet/api/overview/azure/?view=azure-dotnet

Thanks.

**Comment by live1206** [comment]
@vipray-ms Please help to re-record the tests since the api-version has been changed from preview to stable.

**Comment by vipray-ms** [comment]
@live1206 Done, Added new Commit with new records

> @vipray-ms Please help to re-record the tests since the api-version has been changed from preview to stable.

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 1]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
     public partial class NewRelicMonitorResourceData : Azure.ResourceManager.Models.TrackedResourceData
     {
         public NewRelicMonitorResourceData(Azure.Core.AzureLocation location) : base (default(Azure.Core.AzureLocation)) { }
-        public Azure.ResourceManager.NewRelicObservability.Models.AccountCreationSource? AccountCreationSource { get { throw null; } set { } }
+        public Azure.ResourceManager.NewRelicObservability.Models.NewRelicObservabilityAccountCreationSource? AccountCreationSource { get { throw null; } set { } }
         public Azure.ResourceManager.Models.ManagedServiceIdentity Identity { get { throw null; } set { } }
         public Azure.ResourceManager.NewRelicObservability.Models.LiftrResourceCategory? LiftrResourceCategory { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:60]: I'm not sure if this name `LiftrResourceCategory` will be used by other Liftr services.

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 2]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
...
         public int? LiftrResourcePreference { get { throw null; } }
-        public string MarketplaceSubscriptionId { get { throw null; } }
-        public Azure.ResourceManager.NewRelicObservability.Models.MarketplaceSubscriptionStatus? MarketplaceSubscriptionStatus { get { throw null; } }
-        public Azure.ResourceManager.NewRelicObservability.Models.MonitoringStatus? MonitoringStatus { get { throw null; } }
+        public Azure.Core.ResourceIdentifier MarketplaceSubscriptionId { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:62]: I dont think this is a arm-id

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 3]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
...
     }
-    public partial class AccountResource : Azure.ResourceManager.Models.ResourceData
+    public partial class NewRelicAccountResourceData : Azure.ResourceManager.Models.ResourceData
     {
-        public AccountResource() { }
+        public NewRelicAccountResourceData() { }
         public string AccountId { get { throw null; } set { } }
         public string AccountName { get { throw null; } set { } }
         public string OrganizationId { get { throw null; } set { } }
         public string Region { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:171]: Is this an `azure-location`?

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 4]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
...
     }
-    public partial class AppServiceInfo
+    public partial class NewrelicAppServicesGetContent
     {
-        internal AppServiceInfo() { }
-        public string AgentStatus { get { throw null; } }
-        public string AgentVersion { get { throw null; } }
-        public string AzureResourceId { get { throw null; } }
+        public NewrelicAppServicesGetContent(string userEmail) { }
+        public System.Collections.Generic.IList<string> AzureResourceIds { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:176]: Is this an array of `arm-id`?

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 5]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
...
+        public string UserEmail { get { throw null; } }
     }
-    public partial class AppServicesGetContent
+    public partial class NewrelicHostsGetContent
     {
-        public AppServicesGetContent(string userEmail) { }
+        public NewrelicHostsGetContent(string userEmail) { }
+        public string UserEmail { get { throw null; } }
+        public System.Collections.Generic.IList<string> VmIds { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:183]: Is this an array of `arm-id`?

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 6]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
...
+        public string AccountId { get { throw null; } set { } }
+        public string IngestionKey { get { throw null; } set { } }
+        public string Region { get { throw null; } set { } }
+    }
+    public partial class NewRelicObservabilityAppServiceInfo
+    {
+        internal NewRelicObservabilityAppServiceInfo() { }
+        public string AgentStatus { get { throw null; } }
+        public string AgentVersion { get { throw null; } }
+        public string AzureResourceId { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:242]: Is this an `arm-id`?

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 7]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
...
+        public static implicit operator Azure.ResourceManager.NewRelicObservability.Models.NewRelicObservabilityAccountCreationSource (string value) { throw null; }
+        public static bool operator !=(Azure.ResourceManager.NewRelicObservability.Models.NewRelicObservabilityAccountCreationSource left, Azure.ResourceManager.NewRelicObservability.Models.NewRelicObservabilityAccountCreationSource right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class NewRelicObservabilityAccountInfo
+    {
+        public NewRelicObservabilityAccountInfo() { }
+        public string AccountId { get { throw null; } set { } }
+        public string IngestionKey { get { throw null; } set { } }
+        public string Region { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:235]: Is this an `azure-location`?

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 8]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
...
+    }
+    public partial class NewRelicPlan
+    {
+        public NewRelicPlan() { }
+        public Azure.ResourceManager.NewRelicObservability.Models.NewRelicObservabilityBillingCycle? BillingCycle { get { throw null; } set { } }
+        public System.DateTimeOffset? EffectiveOn { get { throw null; } set { } }
+        public string PlanDetails { get { throw null; } set { } }
+        public Azure.ResourceManager.NewRelicObservability.Models.NewRelicObservabilityUsageType? UsageType { get { throw null; } set { } }
+    }
+    public partial class NewRelicPlanResourceData : Azure.ResourceManager.Models.ResourceData
```
Comment (ArthurMa1978) [inline sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:543]: Avoid to use `Resource` as suffix, so here `NewRelicPlanData` should be better

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 9]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
...
+    public partial class NewRelicOrganizationResourceData : Azure.ResourceManager.Models.ResourceData
+    {
+        public NewRelicOrganizationResourceData() { }
+        public Azure.ResourceManager.NewRelicObservability.Models.NewRelicObservabilityBillingSource? BillingSource { get { throw null; } set { } }
         public string OrganizationId { get { throw null; } set { } }
-        public Azure.ResourceManager.NewRelicObservability.Models.PlanData PlanData { get { throw null; } set { } }
-        public string UserEmail { get { throw null; } }
+        public string OrganizationName { get { throw null; } set { } }
+    }
+    public partial class NewRelicPlan
```
Comment (ArthurMa1978) [inline sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:535]: If you rename `NewRelicPlanResourceData` to `NewRelicPlanData`, then this model should be rename to `NewRelicPlanDetails`

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 10]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
...
-        public string FirstName { get { throw null; } set { } }
-        public string LastName { get { throw null; } set { } }
-        public string PhoneNumber { get { throw null; } set { } }
-    }
-    public partial class VmExtensionPayload
-    {
-        internal VmExtensionPayload() { }
-        public string IngestionKey { get { throw null; } }
+        public NewrelicSwitchBillingContent(string userEmail) { }
+        public string AzureResourceId { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:606]: Is this an `arm-id` ?

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs [hunk 11]
```diff
@@ -55,566 +55,566 @@ public partial class NewRelicMonitorResourceCollection : Azure.ResourceManager.A
...
+        public string OrganizationId { get { throw null; } set { } }
+        public Azure.ResourceManager.NewRelicObservability.Models.NewRelicPlan PlanData { get { throw null; } set { } }
+        public string UserEmail { get { throw null; } }
     }
-    public partial class VmInfo
+    public partial class ResourceMonitoredByNewRelic
```
[the remark this answers is not in the document]
Comment (live1206) [reply sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:611]: Sounds good

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs (hunk 1 above)
Comment (live1206) [reply sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:60]: Yes, it is used by other RPs, updated.

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs (hunk 3 above)
Comment (live1206) [reply sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:171]: Updated

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs (hunk 4 above)
Comment (live1206) [reply sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:176]: Updated

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs (hunk 5 above)
Comment (live1206) [reply sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:183]: Updated

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs (hunk 7 above)
Comment (live1206) [reply sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:235]: Updated

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs (hunk 6 above)
Comment (live1206) [reply sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:242]: Updated

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs (hunk 9 above)
Comment (live1206) [reply sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:535]: Updated

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs (hunk 8 above)
Comment (live1206) [reply sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs:543]: Updated

File: sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs (hunk 10 above)
Comment (live1206) [reply sdk/newrelicobservability/Azure.ResourceManager.NewRelicObservability/api/Azure.ResourceManager.NewRelicObservability.netstandard2.0.cs]: Updated

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**